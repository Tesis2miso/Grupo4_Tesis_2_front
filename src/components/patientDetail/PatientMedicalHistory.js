import Toolbar from '@mui/material/Toolbar';
import Box from '@mui/material/Box';
import List from '@mui/material/List'; import { useTranslation } from "react-i18next";
import { useCallback, useEffect, useState } from 'react';
import { styled } from '@mui/material/styles';
import Paper from '@mui/material/Paper';
import './PatientDetail.css'

import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell, { tableCellClasses } from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import CircularProgress from '@mui/material/CircularProgress';
import { toast } from 'react-toastify';
import axios from 'axios';
import useToken from '../utils/useToken';

function PatientDetail(props) {
  const { t } = useTranslation();
  const [loading, setLoading] = useState(false);
  const [isFirstTime, setIsFirstTime] = useState(true);
  const [consults, setConsults] = useState([]);
  const { getToken } = useToken();
  const StyledTableCell = styled(TableCell)(({ theme }) => ({
    [`&.${tableCellClasses.head}`]: {
      backgroundColor: theme.palette.common.black,
      color: theme.palette.common.white,
    },
    [`&.${tableCellClasses.body}`]: {
      fontSize: 14,
    },
  }));
  const StyledTableRow = styled(TableRow)(({ theme }) => ({
    '&:nth-of-type(odd)': {
      backgroundColor: theme.palette.action.hover,
    },
    '&:last-child td, &:last-child th': {
      border: 0,
    },
  }));
  const getMedicalHistory = useCallback(() => {
    let token = getToken();
    axios({
      method: "GET",
      url: `${process.env.REACT_APP_BASE_PATH}/users/${user.id}/consults`,
      headers: {
        'Authorization': `Bearer ${token}`
      }
    })
      .then((response) => {
        const { data } = response;
        setConsults(data);
        setLoading(false);
      }).catch((error) => {
        let mssg = error.response.data.msg;
        toast(mssg);
        setLoading(false);
      })
  })
  useEffect(() => {
    if (isFirstTime) {
      setIsFirstTime(false)
      getMedicalHistory();
    }
  }, [isFirstTime, getMedicalHistory])

  const user = {
    id: 1,
    created_at: "2022-02-02",
    name: "William",
    email: "drummerwilliam@gmail.com",
    birth_day: "2022-02-02",
    city: "Bogot√°",
    phone: "3013016284"
  }

  return (
    <div>
      {
        loading ? (
          <CircularProgress />
        ) : (
          <TableContainer component={Paper} className="medicalHistoryTable">
            <Table sx={{ minWidth: 650 }} aria-label="caption table" >
              <TableHead>
                <TableRow>
                    <StyledTableCell>{t('patientDetailMedicalHistoryEvent')}</StyledTableCell>
                    <StyledTableCell align="right">{t('patientDetailMedicalHistoryEventDate')}</StyledTableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                <StyledTableRow>
                  <TableCell component="th" scope="row">
                    {t('patientDetailMedicalHistoryAccountCreation')}
                  </TableCell>
                  <TableCell align="right">{user.created_at}</TableCell>
                </StyledTableRow>
                {consults.map((consultation) => (
                  <StyledTableRow key={consultation.id}>
                    <TableCell component="th" scope="row">
                      {t('patientDetailMedicalHistoryConsultOfType')} {consultation.injury_type}
                    </TableCell>
                    <TableCell align="right">
                      {consultation.created_at}
                    </TableCell>
                  </StyledTableRow>
                ))}
              </TableBody>
            </Table >
          </TableContainer >
        )
      }
    </div>
  );
}

export default PatientDetail;
