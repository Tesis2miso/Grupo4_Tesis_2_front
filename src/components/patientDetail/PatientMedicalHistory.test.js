import '@testing-library/jest-dom';
import PatientMedicalHistory from './PatientMedicalHistory';
import axios from "axios";
import { mount, shallow } from 'enzyme';
import { act } from 'react-dom/test-utils';

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));
jest.mock('axios', () => jest.fn());
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => { }),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => { },
  }
}));

describe('Tests_Patient_Medical_History', () => {

  var wrapper = null;
  var consults = null;

  beforeEach(() => {
    consults = [
      {
        "color": "rojo",
        "created_at": "2023-02-13T16:51:04.932372",
        "distribution": "brazo",
        "id": 1.0,
        "injuries_count": 1.0,
        "injury_type": "test",
        "photo_url": "https://google.com/",
        "shape": "circular",
        "user_id": 1.0
      }
    ]
    wrapper = shallow(<PatientMedicalHistory />);
  });

  test('show patient detail', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('load data with error', async () => {
    axios.mockResolvedValueOnce(Promise.reject({ response: { data: { mssg: '' } } }));
    const prop = {
      OnValChange: jest.fn()
    }
    await act(async () => mount(<PatientMedicalHistory {...prop} />));
  })

  test('load data', async () => {
    axios.mockResolvedValueOnce(Promise.resolve({ data: consults }));
    const prop = {
      OnValChange: jest.fn()
    }
    await act(async () => mount(<PatientMedicalHistory {...prop} />));
  })
})