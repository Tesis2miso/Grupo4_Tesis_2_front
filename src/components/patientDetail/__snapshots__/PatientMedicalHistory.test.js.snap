// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests_Patient_Medical_History show patient detail 1`] = `
<div>
  <ForwardRef(TableContainer)
    className="medicalHistoryTable"
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "propTypes": Object {
          "children": [Function],
          "className": [Function],
          "classes": [Function],
          "component": [Function],
          "elevation": [Function],
          "square": [Function],
          "sx": [Function],
          "variant": [Function],
        },
        "render": [Function],
      }
    }
  >
    <ForwardRef(Table)
      aria-label="caption table"
      sx={
        Object {
          "minWidth": 650,
        }
      }
    >
      <ForwardRef(TableHead)>
        <ForwardRef(TableRow)>
          <Styled(ForwardRef(TableCell))>
            patientDetailMedicalHistoryEvent
          </Styled(ForwardRef(TableCell))>
          <Styled(ForwardRef(TableCell))
            align="right"
          >
            patientDetailMedicalHistoryEventDate
          </Styled(ForwardRef(TableCell))>
        </ForwardRef(TableRow)>
      </ForwardRef(TableHead)>
      <ForwardRef(TableBody)>
        <Styled(ForwardRef(TableRow))>
          <ForwardRef(TableCell)
            component="th"
            scope="row"
          >
            patientDetailMedicalHistoryAccountCreation
          </ForwardRef(TableCell)>
          <ForwardRef(TableCell)
            align="right"
          >
            2022-02-02
          </ForwardRef(TableCell)>
        </Styled(ForwardRef(TableRow))>
      </ForwardRef(TableBody)>
    </ForwardRef(Table)>
  </ForwardRef(TableContainer)>
</div>
`;
